lib/ash/type/time.ex

```diff
@@ -0,0 +1,31 @@
+defmodule Ash.Type.Time do
+  @moduledoc """
+  Represents a time in the database
+
+  A builtin type that can be referenced via `:time`
+  """
+  use Ash.Type
+
+  @impl true
+  def storage_type, do: :time
+
+  @impl true
+  def cast_input(value, _) do
```

Comment:
We should add a `def cast_input(nil, _), do: {:ok, nil}`

Reply:
Fixed